backend/substrapp/tasks/tasks.py

```diff
@@ -137,34 +142,43 @@ def find_training_step_tuple_from_key(tuple_key):
         f'Key {tuple_key}: no tuple found for training step: model: {metadata}')
 
 
-def get_and_put_model_content(tuple_type, tuple_key, tuple_, out_model, model_dst_path):
+def get_and_put_model_content(tuple_type, hash_key, tuple_, out_model, model_dst_path):
     """Get out model content."""
     owner = tuple_get_owner(tuple_type, tuple_)
     return get_and_put_asset_content(
         out_model['storageAddress'],
         owner,
         out_model['hash'],
         content_dst_path=model_dst_path,
-        salt=tuple_key
+        hash_key=hash_key
     )
 
 
-def get_and_put_local_model_content(tuple_key, out_model, model_dst_path):
+def get_and_put_local_model_content(hash_key, out_model, model_dst_path):
     """Get local model content."""
     from substrapp.models import Model
 
     model = Model.objects.get(pk=out_model['hash'])
 
     # verify that local db model file is not corrupted
-    if get_hash(model.file.path, tuple_key) != out_model['hash']:
+    def validate_hash(path, key, expected):
+        if get_hash(model.file.path, key) == expected:
+            return True
+
+        if settings.SUPPORT_LEGACY_MODEL_SALT and key.endswith(HASH_SALT_HEAD):
+            key = key[:-len(HASH_SALT_HEAD)]
+            return get_hash(model.file.path, key) == expected
+        else:
+            return False
```

Comment:
You should create `validate_hash` function outside as it will be create each time you call the function `get_and_put_local_model_content` !